### ISSUE TITLE
Displaying deleted files

### ISSUE THREAD

**Issue by adambuczynski** [atom/fuzzy-finder#168]
I assume the fuzzy finder uses some kind of cache for the files, but when I delete a file through Atom, it should probably trigger an event to remove that file from the fuzzy finder's cache.

Currently, after having deleted a file, it will still show up in the results of the fuzzy finder, which is confusing at best. 

Possibly related, but the find & replace module is also plagued by this issue.

**Comment by adambuczynski** [comment]
This one is not related to deleted files staying open in tab.
Simple to reproduce:

1. Create a new file with unique filename
2. Open the fuzzy finder and ensure the file shows up when you search for it
3. Delete the file
4. Close the tab is still open
5. Open fuzzy finder and type the filename again
6. The file still shows up
7. After a little while, when using fuzzy finder again, the file doesn't show anymore.

Seems to be a caching issue.

**Comment by adambuczynski** [comment]
I might also add that files that have been moved to another location still show up in the old location in fuzzy finder for a period of time (several minutes in my case). Not sure what triggers the cache refresh, but I'm thinking interactions with files via tree-view should trigger some events to purge/update that cache.

**Comment by anderoonies** [comment]
another that seems to be linked to atom/node-pathwatcher#104

**Comment by adambuczynski** [comment]
Today I renamed a folder `tests` to `test`. After having done some refactoring (would have been about 20 minutes), I used fuzzy finder to jump to one of the test spec files. It opened a blank tab for me without contents. When I opened the file from the tree view, it showed the correct file contents.

To me this seems to be related to this issue, e.g. fuzzy finder caching the files and file locations, and I think it tried to show me the file in the old location, which didn't exist anymore, hence the blank tab.

If however you prefer me to raise a new issue for this, let me know.

**Comment by acontreras89** [comment]
atom/node-pathwatcher#106 has been merged, which may solve this problem. @adambuczynski can you ensure this is working properly with the latest version of `node-pathwatcher`?

**Comment by adambuczynski** [comment]
@acontreras89 this does not fix this issue I'm afraid. Fuzzy finder seems to index/cache the files, and files that are moved to trash don't seem to get removed from the index.

Only after opening fuzzy finder several times does the result disappear at one stage.
